### PR TITLE
[TinyCBOR] Add new port

### LINKED ISSUE #23465: [New Port Request] tinycbor
Library name: tinycbor

Library description: Concise Binary Object Representation (CBOR) Library

Source repository URL: https://github.com/intel/tinycbor

Project homepage (if different from the source repository):

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by alisonrwu**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #23465

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Windows only; I've only tried installing tinycbor:x86-windows locally myself.
  No, am I supposed to update the CI baseline..? By default the new port should build right, so I've not updated it.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes, if committed the result means using the commit hash in the `versions/t-/tinycbor.json`.

### CODE REVIEW

**Review by ras0219-msft — APPROVED** [review]

File: ports/tinycbor/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+vcpkg_cmake_install()
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+
+# Remove duplicated include headers
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+# Remove empty folders
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/lib ${CURRENT_PACKAGES_DIR}/lib)
+SET(VCPKG_POLICY_DLLS_WITHOUT_LIBS enabled)
+SET(VCPKG_POLICY_DLLS_WITHOUT_EXPORTS enabled)
```
Comment (ras0219-msft) [inline ports/tinycbor/portfile.cmake:22]: These indicate that the library doesn't support building as a DLL. Can you remove both of these lines and instead add the following at the top of the file?

```cmake
vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

**Comment by alisonrwu** [comment]
@ras0219-msft @JonLiu1993 Requested changes should be resolved, lmk if anything else needs addressing!

File: ports/tinycbor/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "tinycbor",
+  "version-semver": "0.6.0",
+  "description": "Concise Binary Object Representation (CBOR) Library",
+  "dependencies": [
```
Comment (JonLiu1993) [inline ports/tinycbor/vcpkg.json:5]:
```suggestion
  "description": "Concise Binary Object Representation (CBOR) Library",
  "homepage": "https://github.com/intel/tinycbor",
  "dependencies": [
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/tinycbor/vcpkg.json (hunk 2 above)
Comment (JonLiu1993) [reply ports/tinycbor/vcpkg.json:5]: @alisonrwu ,We should add "homepage" to vcpkg.json when add a new port

**Review by JonLiu1993 — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**